src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs

```diff
@@ -0,0 +1,332 @@
+                Record = pr,
+                TotalSize = fileInfo.Length
+            };
+
+            SingleTransferContext transferContext = this.GetTransferContext(data);
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+            transferContext.SetAttributesCallbackAsync = async (destination) =>
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+            {
+                CloudBlob destBlob = destination as CloudBlob;
+                SetBlobProperties((CloudBlockBlob)destBlob, this.BlobProperties, false);
+                SetBlobMetaData((CloudBlockBlob)destBlob, this.BlobMetadata, false);
+            };
+
+            await DataMovementTransferHelper.DoTransfer(() =>
+            {
+                return this.TransferManager.UploadAsync(filePath,
+                    blob,
+                    null,
+                    transferContext,
+                    this.CmdletCancellationToken);
+            },
+                data.Record,
+                this.OutputStream).ConfigureAwait(false);
+
+            // Set blob permission with umask, since create Blob API still not support them
+            SetBlobPermissionWithUMask((CloudBlockBlob)blob, this.Permission, this.Umask);
+
+            blob.FetchAttributes();
+            WriteDataLakeGen2Item(localChannel, (CloudBlockBlob)blob, taskId: data.TaskId);
+        }
+
+        /// <summary>
+        /// Set block blob permission with Umask after blob is created
+        /// Add this since XSCL don't have interface to set permission with umask in create blob
+        /// Won't set when both permission and umask are null/empty, and use server default behavior.
+        /// </summary>
+        /// <param name="blob">the blob object to set permission with umask</param>
+        /// <param name="permission">permission string to set, in format "rwxrwxrwx", default value is "rwxrwxrwx"</param>
+        /// <param name="umask">umask string to set, in format "rwxrwxrwx", default value is "----w-rwx"</param>
+        protected static void SetBlobPermissionWithUMask(CloudBlockBlob blob, string permission, String umask)
+        {
+            // Don't need set permission when both input permission and umask are null
+            if (string.IsNullOrEmpty(permission) && string.IsNullOrEmpty(umask))
+            {
+                return;
+            }
+
+            //Set the default value if one of permission or Umask is null
+            // Confirmed with feature team:
+            //  Default permission for files is 666 (default permission for directory is 777, this function is for file so use 666)
+            //  default umask is 027
+            if (string.IsNullOrEmpty(permission))
+            {
+                permission = "rw-rw-rw-";
+            }
+            if (string.IsNullOrEmpty(umask))
+            {
+                umask = "----w-rwx";
```

Comment:
I think "rw-rw-rw-" and "----w-rwx" maybe better to be a contant.

Reply:
Will modify it.